### ISSUE TITLE
Shutting down many agents

### ISSUE THREAD

**Issue by Peque** [opensistemas-hub/osbrain#165]
When a name server tries to shut down many agents, they all make proxies to the name server and the threaded Pyro4 pool gets saturated.

This test reproduces the error:

```python
def test_nameserver_proxy_shutdown_with_many_agents():
    """
    Shutdown a name server from a name server proxy when there are many agents
    registered in the name server (make sure proxies do not saturate the name
    server on shutdown).
    """
    import Pyro4
    Pyro4.config.THREADPOOL_SIZE = 4
    ns = run_nameserver()
    for i in range(20):
        run_agent('Agent%s' % i)
    ns.shutdown()
```

This bug affects our walk-forward exploration architecture.

**Comment by Peque** [comment]
Opened a pull request: #166 (an actual fix is currently missing)

**Comment by Peque** [comment]
Fixed in #166.
